src/cosalib/meta.py

```diff
@@ -56,9 +74,10 @@ def validate(self):
         validate ensures that the meta structure matches the schema
         expected.
         """
-        if not self._validator:
-            return
-        self._validator.validate(dict(self))
+        for k, v in self._validator.items():
+            if k in ("main", self.get("name")):
+                log.debug(f"validating against {k} schema")
+                v.validate(dict(self))
```

Comment:
Hmm, so this is just doing a sequential check of the various schemas, right? But I think one other really valuable part of validating against a schema is erroring/warning against unknown keys. Otherwise, e.g. it would not have prevented https://github.com/coreos/coreos-assembler/issues/1045 really.

Briefly looking through https://python-jsonschema.readthedocs.io/en/stable/, I couldn't find an option to specify multiple schemas to "overlay", which I think is understandable.

How about something like this:
- we read in the OS-specific schema file, and add the `properties` items to the `properties` of the base schema
- we turn on `additionalProperties: false` in the base schema

Reply:
Take another look :)